cfn-resources/search-deployment/cmd/resource/resource.go

```diff
@@ -0,0 +1,221 @@
+	apiReq := newSearchDeploymentReq(currentModel)
+	apiResp, res, err := connV2.AtlasSearchApi.UpdateAtlasSearchDeployment(context.Background(), projectID, clusterName, &apiReq).Execute()
+	if err != nil {
+		return handleSearchDeploymentError(res, err)
+	}
+
+	newModel := newCFNSearchDeployment(currentModel, apiResp)
+	return inProgressEvent("Updating Search Deployment", &newModel), nil
+}
+
+func Delete(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+	util.SetDefaultProfileIfNotDefined(&currentModel.Profile)
+
+	if modelValidation := validator.ValidateModel(DeleteRequiredFields, currentModel); modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	client, progressErr := util.NewAtlasV2OnlyClientLatest(&req, currentModel.Profile, true)
+	if progressErr != nil {
+		return *progressErr, nil
+	}
+	connV2 := client.AtlasSDK
+
+	// handling of subsequent retry calls
+	if _, ok := req.CallbackContext[constants.ID]; ok {
+		return handleStateTransition(*connV2, currentModel, constants.DeletedState), nil
+	}
+
+	projectID := util.SafeString(currentModel.ProjectId)
+	clusterName := util.SafeString(currentModel.ClusterName)
+	if resp, err := connV2.AtlasSearchApi.DeleteAtlasSearchDeployment(context.Background(), projectID, clusterName).Execute(); err != nil {
+		return handleSearchDeploymentError(resp, err)
+	}
+
+	return inProgressEvent(constants.DeleteInProgress, currentModel), nil
+}
+
+func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	return handler.ProgressEvent{}, errors.New("not implemented: List")
+}
+
+func handleStateTransition(connV2 admin.APIClient, currentModel *Model, targetState string) handler.ProgressEvent {
+	projectID := util.SafeString(currentModel.ProjectId)
+	clusterName := util.SafeString(currentModel.ClusterName)
+	apiResp, resp, err := connV2.AtlasSearchApi.GetAtlasSearchDeployment(context.Background(), projectID, clusterName).Execute()
+	if err != nil {
+		if targetState == constants.DeletedState && resp.StatusCode == http.StatusBadRequest && strings.Contains(err.Error(), SearchDeploymentDoesNotExistsError) {
+			return handler.ProgressEvent{
+				OperationStatus: handler.Success,
+				ResourceModel:   nil,
+				Message:         constants.Complete,
+			}
+		}
+		return progressevent.GetFailedEventByResponse(err.Error(), resp)
+	}
+
+	newModel := newCFNSearchDeployment(currentModel, apiResp)
+	if util.SafeString(currentModel.StateName) == targetState {
+		return handler.ProgressEvent{
+			OperationStatus: handler.Success,
+			ResourceModel:   newModel,
+			Message:         constants.Complete,
+		}
+	}
+
+	return inProgressEvent(constants.Pending, &newModel)
+}
+
+// specific handling for search deployment API where 400 status code can include AlreadyExists or DoesNotExist that need specific mapping to CFN error codes
+func handleSearchDeploymentError(res *http.Response, err error) (handler.ProgressEvent, error) {
```

Comment:
```suggestion
func handleError(res *http.Response, err error) (handler.ProgressEvent, error) {
```

as we're inside search-deployment resource, and it's a private function

Reply:
makes sense, adjusted